src/routes/+layout.svelte

```diff
@@ -8,10 +10,17 @@
 	import { store as iapStore } from '$lib/stores/inapp_notifications';
 	import Toast from '$lib/components/Toast.svelte';
 
-	export let data: LayoutData;
+	interface Props {
+		data: LayoutData;
+		children?: import('svelte').Snippet;
+	}
+
+	let { data, children }: Props = $props();
 
-	let { supabase, session, pathname } = data;
-	$: ({ supabase, session, pathname } = data);
+	let { supabase, session, pathname } = $state(data);
+	run(() => {
```

Comment:
Is it not possible to use [`$effect or $derived`](https://svelte.dev/docs/svelte/v5-migration-guide#Reactivity-syntax-changes-$:-$derived-$effect) here?

Reply:
The entire `run` block and import should be removed and we can do
```js
let { supabase, session, pathname } = $derived(data);
```

Reply:
Yup, I started doing this in a bunch of other routes and components.